src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -395,6 +400,56 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
         return requestOptions;
     }
 
+    public <T> List<T> find(@NonNull DocumentQuery query, Class<T> domainClass, @NonNull String collectionName) {
+        Assert.hasText(collectionName, "collection should not be null, empty or only whitespaces");
+
+        final FeedOptions feedOptions = new FeedOptions();
+        final DocumentCollection collection = getDocCollection(collectionName);
+        final Optional<String> partitionKeyName = getPartitionKeyField(domainClass);
+        final QuerySpecGenerator generator = new FindQueueSpecGenerator(domainClass);
+
+        feedOptions.setEnableCrossPartitionQuery(!query.getSubjectCriteria(partitionKeyName.orElse("")).isPresent());
```

Comment:
Move `!query.getSubjectCriteria(partitionKeyName.orElse("")).isPresent()` as a variable?

Reply:
OK, should be align with delete code.